### PR TITLE
- New Adapter: Innity

### PR DESCRIPTION
**Pull request by engtat**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change
Hi, new Innity Prebid.js adapter

- test parameters for validating bids
```
{
  bidder: 'innity',
  params: {
    pub: 267,
    zone: 62546
  }
}
```
- engtat@innity.com
- [x] official adapter submission

## Other information
- [Integration example](https://github.com/engtat/Prebid.js/blob/master/integrationExamples/gpt/pbjs_innity_gpt.html)
- [Bidder's params documentation](https://github.com/engtat/prebid.github.io/blob/master/dev-docs/bidders/innity.md)

Thanks.

### CODE REVIEW

File: test/spec/adapters/innity_spec.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    // FYI: querystringify will perform encoding/decoding
+    var querystringify = require('querystringify');
+
+    var adapter = require('src/adapters/innity');
+    var adLoader = require('src/adloader');
+    var bidmanager = require('src/bidmanager');
+    var utils = require('src/utils');
+	var CONSTANTS = require('src/constants.json');
+
+    window.pbjs = window.pbjs || {};
```
Comment (jaiminpanchal27) [inline test/spec/adapters/innity_spec.js]: use $$PREBID_GLOBAL$$ instead of pbjs in this file.

File: test/spec/adapters/innity_spec.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        if (typeof(pbjs._bidsRequested) === "undefined") {
+            pbjs._bidsRequested = [];
+        }
+        if (typeof(pbjs._adsReceived) === "undefined") {
+            pbjs._adsReceived = [];
+        }
+
+        it('bid request for single placement', function () {
+
+            var params = {
```
Comment (jaiminpanchal27) [inline test/spec/adapters/innity_spec.js]: Format this file properly.

File: adapters.json [hunk 3]
```diff
@@ -18,6 +18,7 @@
     "gumgum",
     "hiromedia",
     "indexExchange",
+	"innity",
```
Comment (jaiminpanchal27) [inline adapters.json]: align "innity"

File: test/spec/adapters/innity_spec.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    var expect = require('chai').expect;
+    var urlParse = require('url-parse');
+    
+    // FYI: querystringify will perform encoding/decoding
+    var querystringify = require('querystringify');
+
+    var adapter = require('src/adapters/innity');
+    var adLoader = require('src/adloader');
+    var bidmanager = require('src/bidmanager');
+    var utils = require('src/utils');
```
Comment (jaiminpanchal27) [inline test/spec/adapters/innity_spec.js]: You are not using utils, CONSTANTS anywhere. Remove them if not required

File: test/spec/adapters/innity_spec.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+	
+	describe('creation of bid url', function () {
+
+        if (typeof(pbjs._bidsReceived) === "undefined") { 
+            pbjs._bidsReceived = [];
+        }
+        if (typeof(pbjs._bidsRequested) === "undefined") {
+            pbjs._bidsRequested = [];
+        }
+        if (typeof(pbjs._adsReceived) === "undefined") {
```
Comment (jaiminpanchal27) [inline test/spec/adapters/innity_spec.js]: Why do you need _adsReceived here ? How are you using this ?

**Comment by engtat** [comment]
@jaiminpanchal27 We have made the necessary changes, thanks.

**Comment by mkendall07** [comment]
LGTM.

**Comment by mkendall07** [comment]
This is merged into master. Please submit a PR to the [docs repo](https://github.com/prebid/prebid.github.io) to add a file for your adapter to the [bidders directory](https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders) so your adapter's params will appear on the [bidders page](http://prebid.org/dev-docs/bidders.html). Thank you for contributing

**Pull request merged**